packages/react-dom/src/events/plugins/ModernSelectEventPlugin.js

```diff
@@ -197,19 +163,17 @@ function extractEvents(
   eventSystemFlags,
   targetContainer,
 ) {
-  const doc = getEventTargetDocument(nativeEventTarget);
+  const eventListenerMap = getEventListenerMap(targetContainer);
   // Track whether all listeners exists for this plugin. If none exist, we do
   // not extract events. See #3639.
   if (
-    // We only listen to TOP_SELECTION_CHANGE on the document, never the
-    // root.
-    !isListeningToEvent(TOP_SELECTION_CHANGE, doc) ||
     // If we are handling TOP_SELECTION_CHANGE, then we don't need to
     // check for the other dependencies, as TOP_SELECTION_CHANGE is only
     // event attached from the onChange plugin and we don't expose an
     // onSelectionChange event from React.
-    (topLevelType !== TOP_SELECTION_CHANGE &&
-      !isListeningToEvents(rootTargetDependencies, targetContainer))
+    topLevelType !== TOP_SELECTION_CHANGE &&
+    !eventListenerMap.has('onSelect') &&
+    !eventListenerMap.has('onSelectCapture')
```

Comment:
Can you explain why this check is equivalent? It's not obvious to me.

Reply:
Rather than going through all the dependencies one by one, we can instead check if we've registered `onSelect` or `onSelectCapture`. The dependencies are both the same, as we listen to the same events for both `onSelect` and `onSelectCapture`.